elastic/apm-server
# fix: Add wrapper around libbeat instrumentation to propagate missing tracer config

## Motivation/summary

This PR addresses the [issue](https://github.com/elastic/apm-server/issues/13597) by adding a thin wrapper around libbeat instrumentation to propagate missing tracer config from elastic-agent via environmental variables.

## Checklist


For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Follow instructions in `Building an Elastic Agent container image with a locally built APM Server`  and `Running an Elastic Agent container with a locally built APM Server` section in `TESTING.md`, update the relevant `agent.monitoring` sections in `elastic-agent.yml` to see the config are propagated down to tracing correctly.

## Related issues

#13597
#11381 
#13514
